### PR TITLE
Config server config provider atomic reload

### PR DESCRIPTION
**Pull request by mzakrzewski**
Should address #704

### CODE REVIEW

File: src/Configuration/test/ConfigServerBase.Test/TestConfigServerConfigurationSource.cs [hunk 1]
```diff
@@ -0,0 +1,16 @@
+﻿using Microsoft.Extensions.Configuration;
```
Comment (TimHess) [inline src/Configuration/test/ConfigServerBase.Test/TestConfigServerConfigurationSource.cs:1]:
```suggestion
// Licensed to the .NET Foundation under one or more agreements.
// The .NET Foundation licenses this file to you under the Apache 2.0 License.
// See the LICENSE file in the project root for more information.

﻿﻿using Microsoft.Extensions.Configuration;
```

**Comment by mzakrzewski** [comment]
I'm not sure what's being flagged as wrong with those failed checks. Can you help me understand them?

**Comment by TimHess** [comment]
> I'm not sure what's being flagged as wrong with those failed checks. Can you help me understand them?
> 
> Also, do you know why it is registering such a drop in coverage?

The copyright header was missing from the new file... I added it via suggestion, but apparently an invisible character somehow snuck in. The reported coverage drop is because the tests didn't run, but the coverage uploader still did... All silly/annoying/confusing little things, not a big deal

File: src/Configuration/test/ConfigServerBase.Test/TestConfigServerConfigurationSource.cs (hunk 1 above)
Comment (TimHess) [reply src/Configuration/test/ConfigServerBase.Test/TestConfigServerConfigurationSource.cs:5]:
```suggestion
using Microsoft.Extensions.Configuration;
```

Comment (TimHess) [reply src/Configuration/test/ConfigServerBase.Test/TestConfigServerConfigurationSource.cs:5]: This is where the extra character is, but I can't seem to apply the change

**Comment by TimHess** [comment]
@mzakrzewski After this branch has been brought up to date with 2.x and the extra character has been removed I'll approve & merge

**Review by TimHess — APPROVED** [review]

**Comment by TimHess** [comment]
Thanks again @mzakrzewski

**Pull request merged**